packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -0,0 +1,244 @@
+     * Whether the placeholder is included as an option to clear selection.
+     */
+    allowClear: PropTypes.bool,
+    /**
+     * Optional override of the default display.
+     */
+    display: PropTypes.string,
+    /**
+     * Optional override of the default value.
+     */
+    value: PropTypes.string,
+  }),
+  /**
+   * Whether the field is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the select. Can be a string or number.
+   */
+  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const defaultProps = {
+  disabled: false,
+  isIncomplete: false,
+  isInvalid: false,
+  options: [],
+  required: false,
+};
+
+const createPlaceholder = (placeholder, intl) => {
+  if (!placeholder) {
+    return undefined;
+  }
+  const display = placeholder.display ? placeholder.display : intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
+  const value = placeholder.value ? placeholder.value : defaultPlaceholderValue;
+  const attrs = placeholder.allowClear ? {} : { disabled: true, hidden: true };
+  return <option value={value} {...attrs}>{display}</option>;
+};
+
+const createOptions = options => {
+  const currentOptGroupKeys = [];
+
+  return options.map(current => {
+    if (current.options) {
+      const optGroupKeyIndex = geOptGroupKeyIndex(current.display, currentOptGroupKeys);
+      const optGroupKey = getOptGroupKey(current.display, optGroupKeyIndex);
+      currentOptGroupKeys.push(optGroupKey);
+
+      return (
+        <optgroup key={optGroupKey} label={current.display}>
+          {createOptions(current.options)}
+        </optgroup>
+      );
+    }
+    return <option key={`${current.value}`} value={current.value}>{current.display}</option>;
+  });
+};
+
+const NativeSelect = ({
+  ariaDescribedBy,
+  ariaLabel,
+  attrs,
+  disabled,
+  defaultValue,
+  id,
+  intl,
+  isInvalid,
+  isIncomplete,
+  onChange,
+  options,
+  placeholder,
+  required,
+  value,
+  ...customProps
+}) => {
+  const [currentValue, setCurrentValue] = useState(defaultValue || getFirstValue(options, placeholder));
+  const refIsControlled = useRef(!defaultValue && value);
+  const refSelect = useRef();
+  const theme = React.useContext(ThemeContext);
+
+  const handleOnMouseDown = () => {
+    refSelect.current.setAttribute('data-focus-interaction', 'mouse');
+  };
+  const handleOnBlur = () => {
+    refSelect.current.setAttribute('data-focus-interaction', 'none');
+  };
+  const handleOnFocus = () => {
+    if (refSelect.current.getAttribute('data-focus-interaction') !== 'mouse') {
+      refSelect.current.setAttribute('data-focus-interaction', 'keyboard');
+    }
+  };
+  const handleOnChange = event => {
+    if (onChange) {
+      onChange(event);
+    }
+    if (!refIsControlled.current) {
+      setCurrentValue(event.currentTarget.value);
+    }
+  };
+
+  let controlledValue;
+  if (refIsControlled.current) {
+    controlledValue = value || getFirstValue(options, placeholder);
```

Comment:
I think this can be just ` controlledValue = value `. Since we are validating `const refIsControlled = useRef(!defaultValue && value);` I don't think value would be null when `if (refIsControlled.current)` validates to true and also we have one more check on line no. 192 where if `controlledValue` is null / undefined it will be replaced with `currentValue` ( which will be either `defaultValue` or `getFirstValue` )

Reply:
> if i user clears the selected value the select should reset to the first item.

Correct me if I am wrong. it resets to first item in line No. 192 also : ` value: controlledValue || currentValue`

Reply:
That would only be correct if the list of options stayed static, as the default is only captured on initial mount, if the user adds/removes items in the select there is a potential an incorrect value would attempt to be restored, so the value needs to be captured anew is the value is cleared. There are other potential ways to account for this, but it is covered with this logic.

I can potentially update the state variable to `uncontrolledValue` rather that `currentValue` to make it clearer.

Reply:
I updated the component to be a little more explicit about how it manages values.